fallback to uid when no username

zdzichu opened the pull request:
##### SUMMARY
When running ansible in a container, with `delegate_to: localhost`, it is possible the container is run as a random UID without corresponding entry in `/etc/passwd`. In such situation `getpass.getuser()` will throw an exception:
`An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'getpwuid(): uid not found: 1000'`

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
Connection plugin: local

##### ADDITIONAL INFORMATION
Identical fix was done in `display.py`: https://github.com/ansible/ansible/pull/68521/commits/0a1d8835605f139ba3c4cb66aba1feef33d61790

There are few more places where ansible calls `getpass.getuser()`, but they do not need this fix.

Akasurde commented on lib/ansible/plugins/connection/local.py:
```diff
@@ -47,7 +47,11 @@ def __init__(self, *args, **kwargs):
         self.cwd = None
-        self.default_user = getpass.getuser()
+        try:
+            self.default_user = getpass.getuser()
```
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add tests for this change.

bcoca replied:
we use getpass.getuser() in many places, we might want to make this a generic function in module_utils and replace the current  instances of it, instead of fixing just this one

zdzichu committed:
fallback to uid when no username

Adds helper function wrapping getpass.getuser(), which
synthetises username when there is none in /etc/passwd.

zdzichu committed:
replace bare getpass.getuser() calls with safer get_username() wrapper

get_username() deals with situations when there is no corresponding entry
in /etc/passwd, which may happen with containers.

In 0a1d8835605f139ba3c4cb66aba1feef33d61790, workaround was applied to
single sitaution. This change extends the fix to other callsites.

zdzichu commented:
Hey, I've fixed the CI errors, can this PR be tested and (eventually) merged?

VinayBalpande commented:
Hello,
This is useful change. Any chance this could be progressed further please?

bcoca commented:
aside from now having a conflict, it needs to pass CI 

waiting_on_contributor
